index.js

```diff
@@ -13,38 +13,57 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-function VWOAmplitudePlugin(amplitude){
-    window.VWO = window.VWO || []
-    window.VWO.push([
-        "onVariationApplied",
-        function (data) {
-            if (!data) return;
-            const expId = data[1];
-            const variationId = data[2];
-            const _vis_data = {};
-            if (
-                expId &&
-                variationId &&
-                ["VISUAL_AB", "VISUAL", "SPLIT_URL"].indexOf(window._vwo_exp[expId].type) > -1
-            ) {
-                _vis_data["VWO-Test-ID-" + expId] = window._vwo_exp[expId].comb_n[variationId];
-                const key = "VWO-Test-ID-" + expId;
-                if (amplitude) {
-                    let identify = new amplitude.Identify();
-                    identify.set(key, _vis_data[key]);
-                    if (amplitude.getInstance) {
-                        amplitude.getInstance().identify(identify);
-                        amplitude.getInstance().logEvent("VWO", _vis_data);
-                    } else {
-                        amplitude.identify(identify);
-                        amplitude.logEvent("VWO", _vis_data);
-                    }
-                } else {
-                    console.warn("VWO Amplitude Plugin Log - amplitude is not defined")
-                }
-            }
-        },
-    ])
+function VWOAmplitudePlugin(amplitude, options = {}) {
+  const { useLegacyKeys = true } = options;
+
+  window.VWO = window.VWO || [];
+  window.VWO.push([
+    "onVariationApplied",
+    function (data) {
+      if (!data || !amplitude) {
+        if (!amplitude) {
+          console.warn("VWO Amplitude Plugin Log - amplitude is not defined");
+        }
+        return;
+      }
+
+      const expId = data[1];
+      const variationId = data[2];
+      const _vis_data = {};
+      if (
+        expId &&
+        variationId &&
+        window._vwo_exp &&
+        window._vwo_exp[expId] &&
+        ["VISUAL_AB", "VISUAL", "SPLIT_URL"].indexOf(
+          window._vwo_exp[expId].type
+        ) > -1
+      ) {
+        let key = "VWO-Test-ID-";
+        const testKeyId = "VWO-Test-ID";
+        const variationKeyId = "VWO-Variation-ID";
+        if (useLegacyKeys) {
+          key += expId;
+          _vis_data[key] = window._vwo_exp[expId].comb_n[variationId];
+        } else {
+          _vis_data[testKeyId] = expId;
+          _vis_data[variationKeyId] =
+            window._vwo_exp[expId].comb_n[variationId];
+        }
+
+        const identify = new amplitude.Identify();
+        if (useLegacyKeys) {
+          identify.set(key, _vis_data[key]);
+        } else {
+          identify.set(testKeyId, _vis_data[testKeyId]);
```

Comment:
This will override the values because the attribute can only store a single value at a time, resulting in the user having only one test value at any given time, which can skew the reports.